src/main/java/org/broadinstitute/hellbender/utils/Histogram.java

```diff
@@ -110,7 +111,7 @@ public Double median() {
 
         int counter = 0;
         Double firstMedian = null;
-        for(final Integer key : dataList.valueCounts.keySet()) {
+        for(final Integer key : new TreeSet<>(dataList.valueCounts.keySet())) {
```

Comment:
@kcibul Can you explain why no integration test results changed as a result of this patch (for example, the HaplotypeCaller allele-specific annotation tests)?

Reply:
Also, is there a similar issue in the `CompressedDataList.iterator()` method?

Reply:
This is a great question -- maybe @ldgauthier can help us out here.  This median method is only called in the `AS_RankSumTest.finalizeRawData` method.  When I search for usages of that method, which is part of the ReducibleAnnotation interface, I don't find any specific usages.  

As for Integration Tests... I'm not sure.  It would have to be an integration test where allele specific annotations are combined, which I think is just Gnarly.  Looking at the GnarlyGenotyperIntegrationTest, it looks like the only test that is Allele Specific uses two samples, which isn't broken since the median of two values is always just the average of the two values regardless of ordering.

Reply:
As for CompressedDataList.. yep that's broken also.  Interestingly the toString() method sorts the data before returning it... but the iterator does not.  I added a test for it, asserting correct ordering and it fails and then applied the same fix